common/src/main/scala/explore/GraphQLSchemas.scala

```diff
@@ -64,4 +66,21 @@ object GraphQLSchemas {
         explore.model.encoders.siderealTargetEncoder
     }
   }
+
+  @GraphQLSchema(debug = false)
+  object UserPreferencesDB {
+    object Scalars {
+      type UserId        = User.Id
+      type ResizableArea = String
+    }
+
+    object Types {
+      type WidthInsertInput = ExploreResizableWidthInsertInput
+      type WidthInsertInputL = Array[ExploreResizableWidthInsertInput]
```

Comment:
Why `Array` instead of `List` ?

Reply:
Just testing, either Array or List produce an exception during macro expansion

Reply:
Might be a bug, what's the exception?

Reply:
I get a very long exception

```
[error] /Users/cquiroz/Projects/explore/common/src/main/scala/explore/common/UserPreferencesQueries.scala:31:4: java.util.NoSuchElementException: None.get
[error]  at scala.None$.get(Option.scala:627)
[error]  atscala.None$.get(Option.scala:626)
[error]  atclue.macros.GraphQLImpl.resolveVariables(GraphQL.scala:236)
[error]  atclue.macros.GraphQLImpl.$anonfun$addVars$1(GraphQL.scala:273)
[error]  atscala.Function$.$anonfun$chain$2(Function.scala:23)
[error]  atscala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:169)
[error]  atscala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:165)
[error]  atscala.collection.immutable.List.foldLeft(List.scala:79)
[error]  atscala.Function$.$anonfun$chain$1(Function.scala:23)
[error]  atclue.macros.GraphQLImpl.$anonfun$expand$6(GraphQL.scala:462)
[error]  atcats.effect.internals.IORunLoop$.step(IORunLoop.scala:241)
[error]  atcats.effect.IO.unsafeRunTimed(IO.scala:338)
[error]  atcats.effect.IO.unsafeRunSync(IO.scala:256)
[error]  atclue.macros.Macro.macroResolve(Macro.scala:19)
[error]  atclue.macros.Macro.macroResolve$(Macro.scala:17)
[error]  atclue.macros.GraphQLImpl.macroResolve(GraphQL.scala:27)
[error]  atclue.macros.GraphQLImpl.resolve(GraphQL.scala:31)
```
and more

Reply:
Thanks, I'll take a look

Reply:
I see the exception pops up during `@GraphQL` expansion, when resolving query variables. This is possible due to the fact that only types defined in the schema (or base types) can be used as query variables. The query is compiled and validated against the schema at compile time by literally looking for type definitions, and so type aliases can't be used.

There is also a bug in `clue` when reporting this kind of issues, which I will address.

Reply:
Thanks for the explanation. I managed to make it work.